src/Workspaces/MSBuild/Test/VisualStudioMSBuildWorkspaceTests.cs

```diff
@@ -3276,6 +3276,33 @@ public async Task TestInvalidSolutionFilterDoesNotLoad()
         Assert.Equal(0, workspace.CurrentSolution.ProjectIds.Count);
     }
 
+    [ConditionalFact(typeof(VisualStudioMSBuildInstalled))]
+    public async Task TestValidXmlSolutionSupport()
+    {
+        CreateFiles(GetMultiProjectSolutionFiles()
+            .WithFile(@"CSharpXmlSolution.slnx", Resources.XmlSolutionFiles.CSharp));
+        var solutionFilePath = GetSolutionFileName(@"CSharpXmlSolution.slnx");
+
+        using var workspace = CreateMSBuildWorkspace();
+        var solution = await workspace.OpenSolutionAsync(solutionFilePath);
+        var csharpProject = solution.Projects.Single();
+
+        Assert.Equal(LanguageNames.CSharp, csharpProject.Language);
```

Comment:
nit: might be nice to assert on its name/path also. Up to you.

Reply:
Added a few more assertions.